lib/cookbook-release/release.rb

```diff
@@ -0,0 +1,107 @@
+class Release
+
+  attr_reader :git
+
+  def initialize(git)
+    @git = git
+  end
+
+  def last_release
+    @last_release ||= git.compute_last_release
+  end
+
+  def git_changelog
+    @git_changelog ||= git.compute_changelog(last_release)
+  end
+
+  # return the new version and the reasons
+  def new_version
+    case true
+    when major_changes?
+      [ last_release.major!, major_changes ]
+    when minor_changes?
+      [ last_release.minor!, minor_changes ]
+    when patch_changes?
+      [ last_release.patch!, patch_changes ]
+    else
+      raise "No commit since last release (#{last_release})"
+    end
+  end
+
+  %w(major minor patch).each do |level|
+    define_method("#{level}_changes?") do
+      (send "#{level}_changes").size > 0
+    end
+
+    define_method("#{level}_changes") do
+      git_changelog.select(&"#{level}?".to_sym)
+    end
+  end
```

Comment:
It would suggest to just compute the changes in the `new_version` method and return directly these changes:

```ruby
def new_version
  %w(major minor patch).each do |level|
    changes = git_changelog.select(&"#{level}?".to_sym)
    return [ last_release.send("#{level}!".to_sym), changes] unless changes.empty?
  end
end
```